iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -409,8 +411,12 @@ public void preAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTable, E
           context.getProperties().get(OLD_TABLE_NAME)).toString());
     }
     if (commitLock == null) {
-      commitLock = new MetastoreLock(conf, new CachedClientPool(conf, Maps.fromProperties(catalogProperties)),
-          catalogProperties.getProperty(Catalogs.NAME), hmsTable.getDbName(), hmsTable.getTableName());
+      if (HiveTableOperations.hiveLockEnabled(null, conf)) {
```

Comment:
HiveIcebergMetaHook should always use MetastoreLock 
PS: `iceberg-catalog` could be dropped once iceberg Hive-4.0 support is merged)

Reply:
what is interesting is that I can only see this whole hiveLockEnabled(...) thing used in testing scenarios, which applies to my case as well, see screenshot 
<img width="868" alt="Screenshot 2025-03-10 at 10 40 58" src="https://github.com/user-attachments/assets/8947c5c2-827e-4e72-a8be-32b03574a719" />

this makes me think that if I want to make my query history PR snapshot work then
a) I have to override the locking behavior and bypass it in case of tests by extending the meta hook instead and use a testing one
b) or make it work with metastore locking

Reply:
seems like b) works, I'm about to close this one soon

now I would appreciate a review on https://github.com/apache/hive/pull/5666 to finalize the first phase of query history maybe, thanks in advance!